app/test_main.py

```diff
@@ -1 +1,43 @@
-# write your code here
+from unittest import mock
+import pytest
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+@pytest.mark.parametrize(
+    "connection, url, result",
+    [
+        pytest.param(
+            True,
+            True,
+            "Accessible",
+            id="should return `Accessible` when all is ok"
+        ),
+        pytest.param(
+            False,
+            True,
+            "Not accessible",
+            id="should return `Not accessible` when lost connection"
+        ),
+        pytest.param(
+            True,
+            False,
+            "Not accessible",
+            id="should return `Not accessible` when url is false"
+        )
+    ]
+)
+def test_can_access_google_page(
+        mocked_has_internet_connection: mock.MagicMock,
+        mocked_valid_google_url: mock.MagicMock,
```

Comment:
The order of mocks in your test function's parameters is incorrect relative to the order in which the @mock.patch decorators are applied.